SU-HSDO/suhumsci
# STN-703  Hotfix caption credit bug

# READY FOR REVIEW

## Summary
As a logged out user the caption/credits styling appears in the middle of the screen.

JIRA Card: 
https://sparkbox.atlassian.net/browse/STN-703

Prod Example: 
https://sparkbox-sandbox-prod.stanford.edu/news/test-news-0

Approach:
What I figured out from examining this issue is this is happening from the Media Image fields on node pages.
We don't see the issue when we are logged in because there is a "contextual region" class that is applied for the editor links and that container has `position: relative` applied to it, thus containing the caption/credit div. When logged out that container disappears and so does the position.

As a temporary fix, in this work, I've added `position: relative `to the field classes that are affected. The intention is to remove this work after a more elegant solution has been created using a twig template for the caption-credit media field only, allowing us to add a position: relative wrapper to that application and not apply it to the other view modes the media images can utilize.

## Need Review By (Date)
asap

## Urgency
high

## Steps to Test
1. npm run test
2. Create a test page that uses one of the media image fields for uploading or using images from the media library, such as: event, event series, news, project ct, publication, or person.
![image](https://user-images.githubusercontent.com/9355698/109857802-12739480-7c29-11eb-9be2-f79747926fa4.png)

3. Add an image to that test page that has a caption/credit added to it via the media caption/credit field for the image.
![image](https://user-images.githubusercontent.com/9355698/109857889-29b28200-7c29-11eb-9079-e30719ad8e73.png)

4. Ensure that the content type is displaying the caption/credit view mode of the media image via the Manage Display option for the content type.
![image](https://user-images.githubusercontent.com/9355698/109858141-739b6800-7c29-11eb-8931-9f4e24295d94.png)


5. Verify that the caption credit is appearing correctly as an overlay to the image in both the logged in and logged out state using Incognito browsing for logged out.

Before:
![image](https://user-images.githubusercontent.com/9355698/109857614-e0fac900-7c28-11eb-839f-f53c88ccb44a.png)

After:
![image](https://user-images.githubusercontent.com/9355698/109857739-0091f180-7c29-11eb-8132-46bde5763bef.png)


## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Sparkbox PR Checklist](../docs/SparkboxPRChecklist.md)
